File: Python/0485_max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        running_count, max_count = 0, -1

        for num in nums + [0]:
            max_count = max(running_count, max_count)

            if num == 0:
                running_count = 0
            else:
                running_count += 1

        return max_count

File: Elixir/0485_max-consecutive-ones.ex
defmodule Solution do
  @spec find_max_consecutive_ones(nums :: [integer]) :: integer
  def find_max_consecutive_ones(nums) do
    {max_ones, _} =
      Enum.reduce(nums, {0, 0}, fn num, {max_ones, current_ones} ->
        if num == 1, do: {max(max_ones, current_ones + 1), current_ones + 1}, else: {max_ones, 0}
      end)

    max_ones
  end
end
